samcli/commands/check/resources/LambdaFunction.py

```diff
@@ -0,0 +1,84 @@
+"""
+Class object for Lambda Functions. Contains object from template, as well as all data for lambda functions,
+excluding pricing info
+"""
+from typing import List
+from samcli.lib.providers.provider import Function
+from samcli.commands.check.resources.TemplateResource import TemplateResource
+
+
+class LambdaFunction(TemplateResource):
```

Comment:
It seems a lot of the properties are directly mapping to class members with both get and set.
 
Going back to the previous comment: https://github.com/aws/aws-sam-cli/pull/3109#discussion_r675202180
If the members are expected to be set and get without checks, we can leave them as public members and switch to properties when needed in the future. https://stackoverflow.com/a/6618184/5284278